src/app/backend/handler/apihandler.go

```diff
@@ -93,6 +102,42 @@ func wsMetrics(req *restful.Request, resp *restful.Response, chain *restful.Filt
 	}
 }
 
+// Post requests should set correct X-CSRF-TOKEN header, all other requests
+// should either not edit anything or be already safe to CSRF attacks (PUT
+// and DELETE)
+//
+// Validation handlers are also idempotent functions, and not actual data
+// modification opperations
+func shouldDoCsrfValidation(req *restful.Request) bool {
+	if req.Request.Method != "POST" {
+		return false
+	}
+	if strings.HasPrefix(req.SelectedRoutePath(), "/api/v1/appdeployment/validate/") {
+		return false
+	}
+	return true
+}
+
+func xsrfValidation(csrfKey string) func(*restful.Request, *restful.Response, *restful.FilterChain) {
+
+	return func(req *restful.Request, resp *restful.Response, chain *restful.FilterChain) {
+		resource := mapUrlToResource(req.SelectedRoutePath())
+		if resource == nil || (shouldDoCsrfValidation(req) &&
+			!xsrftoken.Valid(req.HeaderParameter("X-CSRF-TOKEN"),
+				csrfKey,
+				"none",
+				*resource)) {
+
+			err := errors.New("CSRF validation failed")
+			log.Print(err)
```

Comment:
Should we print the error here? Wouldn't it be printed by our error handlers anyway?